hathor/transaction/block.py

```diff
@@ -110,21 +110,27 @@ def calculate_min_height(self) -> int:
 
         return min_height
 
-    def calculate_feature_activation_bit_counts(self) -> list[int]:
+    def get_feature_activation_bit_counts(self) -> list[int]:
         """
-        Calculates the feature_activation_bit_counts metadata attribute, which is a list of feature activation bit
-        counts.
+        Lazily calculates the feature_activation_bit_counts metadata attribute, which is a list of feature activation
+        bit counts. After it's calculated for the first time, it's persisted in block metadata and must not be changed.
 
         Each list index corresponds to a bit position, and its respective value is the rolling count of active bits
         from the previous boundary block up to this block, including it. LSB is on the left.
         """
+        metadata = self.get_metadata()
+
+        if metadata.feature_activation_bit_counts is not None:
+            return metadata.feature_activation_bit_counts
+
         previous_counts = self._get_previous_feature_activation_bit_counts()
         bit_list = self._get_feature_activation_bit_list()
 
         count_and_bit_pairs = zip_longest(previous_counts, bit_list, fillvalue=0)
         updated_counts = starmap(add, count_and_bit_pairs)
+        metadata.feature_activation_bit_counts = list(updated_counts)
```

Comment:
Should you save it?

Reply:
Maybe it would be ideal, but I can't do that here as this method is called in verification context, so I would have the same problem described in https://github.com/HathorNetwork/hathor-core/pull/785#discussion_r1377035035.

It is going to be indirectly saved though, as the metadata is saved by the manager after the verification process is complete.

Reply:
I will test this to make sure it's saved, actually.

Reply:
Got it. Thanks!